controllers/awscluster_controller.go

```diff
@@ -135,6 +135,10 @@ func reconcileNormal(clusterScope *scope.ClusterScope) (reconcile.Result, error)
 	// If the AWSCluster doesn't have our finalizer, add it.
 	if !util.Contains(awsCluster.Finalizers, infrav1.ClusterFinalizer) {
 		awsCluster.Finalizers = append(awsCluster.Finalizers, infrav1.ClusterFinalizer)
+		// Register the finalizer immediately to avoid orphaning AWS resources on delete
+		if err := clusterScope.PatchObject(); err != nil {
+			return reconcile.Result{}, err
+		}
```

Comment:
Do we want/need to return after here and requeue? 

/cc @ncdc

Reply:
The PatchHelper does not mutate the object being patched in memory, so any changes such as to resourceVersion don't come through. We could patch to add the finalizer and return immediately (no requeue). Because it's a change, the object would be added to the queue again fairly quickly.

On the other hand, it's _probably_ safe not to do that, and to just continue with what we already have in memory, as the deferred patch on scope closure will only be patching things we changed in this reconcile iteration.

Reply:
@aaroniscode have you tested these changes?

Reply:
@ncdc yes, I've tested quite a bit and re-tested today. Creating and deleting the cluster no longer orphans AWS resources. And there doesn't appear to be any issue with calling PatchHelper and then continuing with the reconcile.

But I am not an expect on controller runtime and kubebuilder and if there is a potential issue with by not immediately returning and requeuing like @vincepri suggests. Let me know if you think I should consider that route. Thanks!

Reply:
If you haven't run into issues, we can proceed. If we encounter conflict errors in the future, it potentially may come from here (but I think we're ok).